python/pyspark/ml/torch/tests/test_ssh_env_manager.py

```diff
@@ -0,0 +1,134 @@
+import contextlib
+from contextlib import contextmanager
+import os
+import unittest
+import shutil
+from pyspark.ml.torch import ssh_manager
+
+from pyspark.ml.torch.ssh_manager import SSHEnvManager
+from ..utils import write_to_location
+
+KNOWN_HOSTS_TEST_TEMP = "/root/.ssh/known_hosts_before_test"
+KNOWN_HOSTS = "/root/.ssh/known_hosts"
+PRIVATE_SSH_KEY = "private test key here"
+PUBLIC_SSH_KEY = "public test key here"
+class SSHTestEnvSetup():
+    def __init__(self):
+        self.save_known_hosts() 
+    
+    def save_known_hosts(self):
+        if os.path.exists(KNOWN_HOSTS):
+            shutil.move(KNOWN_HOSTS, KNOWN_HOSTS_TEST_TEMP)
+    
+    def restore_known_hosts(self):
+        if os.path.exists(KNOWN_HOSTS_TEST_TEMP):
+            shutil.move(KNOWN_HOSTS_TEST_TEMP, KNOWN_HOSTS)
+
+    @contextmanager 
+    def create_ssh_key_file(self, ssh_key_path: str):
+        if not os.path.exists(ssh_key_path):
+            with open(ssh_key_path, "w+") as f:
+                f.write(PRIVATE_SSH_KEY)
+            with open(f"{ssh_key_path}.pub", "w+") as f:
+                f.write(PUBLIC_SSH_KEY)
+            yield
+            os.remove(ssh_key_path)
+            os.remove(f"{ssh_key_path}.pub")
+
+    @contextmanager
+    def temp_remove_ssh_key_file(self, ssh_key_path: str):
+        if os.path.exists(ssh_key_path):
+            shutil.move(ssh_key_path, f"{ssh_key_path}.temp")
+        if os.path.exists(f"{ssh_key_path}.pub"):
+            shutil.move(f"{ssh_key_path}.pub", f"{ssh_key_path}.pub.temp")
+        yield
+        if os.path.exists(f"ssh_key_path.temp"):
+            shutil.move(f"{ssh_key_path}.temp", ssh_key_path)
+        if os.path.exists(f"{ssh_key_path}.pub.temp"):
+            shutil.move(f"{ssh_key_path}.pub.temp","{ssh_key_path}.pub")
+    
+    @contextmanager
+    def create_known_hosts(self, msg=""):
+        write_to_location("/root/.ssh/known_hosts", msg)
+        yield 
+        os.remove("/root/.ssh/known_hosts")
+
+class SSHEnvManagerUnitTest(unittest.TestCase):
+   
+    def create_ssh_key_test(self):
+       ssh_test_env = SSHTestEnvSetup()
```

Comment:
If you're doing the same setup every time you can use setup[Class] https://docs.python.org/3/library/unittest.html#unittest.TestCase